src/Test/modellike.jl

```diff
@@ -386,6 +386,87 @@ function failcopytestca(dest::MOI.ModelLike)
 end
 
+function start_values_unset_test(model::MOI.ModelLike, config)
+    MOI.empty!(model)
+    x, y, z = MOI.add_variables(model, 3)
+    fz = MOI.SingleVariable(z)
+    a = MOI.add_constraint(model, x, MOI.EqualTo(1.0))
+    b = MOI.add_constraint(model, y, MOI.EqualTo(2.0))
+    F1 = MOI.SingleVariable
+    S1 = MOI.EqualTo{Float64}
+    c = MOI.add_constraint(model, MOIU.operate(vcat, Float64, 2.0fz + 1.0), MOI.Nonnegatives(1))
+    F2 = MOI.VectorAffineFunction{Float64}
+    S2 = MOI.Nonnegatives
+
+    vpattr = MOI.VariablePrimalStart()
+    cpattr = MOI.ConstraintPrimalStart()
+    cdattr = MOI.ConstraintDualStart()
+
+    @testset "Newly created variables have start values set to nothing" begin
+        @test !(vpattr in MOI.get(model, MOI.ListOfVariableAttributesSet()))
+        @test MOI.get(model, vpattr, x) === nothing
+        @test MOI.get(model, vpattr, y) === nothing
+        @test MOI.get(model, vpattr, z) === nothing
+
+        @test !(cpattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}()))
+        @test MOI.get(model, cpattr, a) === nothing
+        @test MOI.get(model, cpattr, b) === nothing
+        @test !(cpattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}()))
+        @test MOI.get(model, cpattr, c) === nothing
+
+        @test !(cdattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}()))
+        @test MOI.get(model, cdattr, a) === nothing
+        @test MOI.get(model, cdattr, b) === nothing
+        @test !(cdattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}()))
+        @test MOI.get(model, cdattr, c) === nothing
+    end
+
+    @testset "Allows unsetting by nothing" begin
+        # First set the attributes to some values.
+        MOI.set.((model,), (vpattr,), (x, y, z), (0.0, 1.0, 2.0))
+        MOI.set.((model,), (cpattr,), (a, b), (0.0, 1.0))
+        MOI.set.((model,), (cdattr,), (a, b), (0.0, 1.0))
+        MOI.set(model, cpattr, c, [2.0])
+        MOI.set(model, cdattr, c, [2.0])
+
+        @test vpattr ∈ MOI.get(model, MOI.ListOfVariableAttributesSet())
+        @test cpattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}())
+        @test cpattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}())
+        @test cdattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}())
+        @test cdattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}())
+
+        @test all(MOI.get.((model,), (vpattr,), (x, y, z)) .== (0.0, 1.0, 2.0))
+        @test all(MOI.get.((model,), (cpattr,), (a, b)) .== (0.0, 1.0))
+        @test all(MOI.get.((model,), (cdattr,), (a, b)) .== (0.0, 1.0))
+        @test MOI.get(model, cpattr, c) == [2.0]
+        @test MOI.get(model, cdattr, c) == [2.0]
+
+        MOI.set(model, vpattr, y, nothing)
+        MOI.set(model, cpattr, a, nothing)
+        MOI.set(model, cdattr, a, nothing)
+        MOI.set(model, cpattr, c, [nothing])
```

Comment:
I would expect it to be `nothing`, not `[nothing]`

Reply:
I will check this. I noticed now that I do not query the `cpattr` of `c` before setting it the first time. If the initial value of `cpattr` for `c` is `[nothing]`, then it would make sense keep it that way, to avoid breaking the reasoning that initially motivated this behavior, that is, ``if variables with an attribute unset return `nothing` when queried, then setting them with the returned value (i.e., nothing) should restore the original/unset state''. 

I have to admit I did not deal directly with this kinda of constraint, so I do not know what to expect here.
```julia
c = MOI.add_constraint(model, MOIU.operate(vcat, Float64, 2.0fz + 1.0), MOI.Nonnegatives(1))
```

Reply:
Oh, okay, I was already checking it before and it has value `nothing`. So the `MOI.set` should take `nothing` and `MOI.get` should return `nothing`. I changed the tests and it works. Should we have some tests for `nothing` values inside the arrays that may be used to set this attribute, or setting the whole constraint to `nothing` and getting `nothing` back is enough?

Reply:
Just changed the tests to expect `nothing` and set `nothing`. No test for `nothing` elements inside a Vector.

Reply:
It's ok if `nothing` element inside vector for a vector constraint is undefined behavior for now, this is out of the scope of this PR.